pkg/csi_driver/controller.go

```diff
@@ -601,9 +604,6 @@ func (s *controllerServer) generateNewFileInstance(name string, capBytes int64,
 			return nil, fmt.Errorf("invalid parameter %q", k)
 		}
 	}
-	if kmsKeyName != "" && tier != enterpriseTier {
```

Comment:
why not add a (tier != enterpriseTier && tier != HighScaleTier) ?

Reply:
this way we can avoid future potential misalignments with the Filestore API. for example if Filestore change what tier has cmek capability we don't have to change our code anymore.

Reply:
Ack. But if user tries out KMS with non compatible tiers, we would create SLO noise

Reply:
[isUserError()](https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/blob/master/pkg/cloud_provider/file/file.go#L649) should ideally handle it. 

Please verify manually if it fails for a incompatible tier and capture the logs in this PR for future reference

Reply:
done